base/applications/mscutils/eventvwr/eventvwr.c

```diff
@@ -3381,16 +3381,42 @@ WndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
                         if ( (pnmv->uChanged  & LVIF_STATE) && /* The state has changed */
                              (pnmv->uNewState & LVIS_SELECTED) /* The item has been (de)selected */ )
                         {
-                            if (hwndEventDetails)
-                                SendMessageW(hwndEventDetails, EVT_DISPLAY, 0, 0);
+                            if (!hwndEventDetails)
+                                break;
+
+                            /* Verify the index of selected item */
+                            if (pnmv->iItem == -1)
+                            {
+                                MessageBoxW(hWnd,
+                                            L"No selected items!",
+                                            szTitle,
+                                            MB_OK | MB_ICONERROR);
+                                break;
+                            }
+                            SendMessageW(hwndEventDetails, EVT_DISPLAY, 0, (LPARAM)pnmv->iItem);
                         }
                         break;
                     }
 
+                #ifdef LVN_ITEMACTIVATE
+                    case LVN_ITEMACTIVATE:
+                #else
                     case NM_DBLCLK:
                     case NM_RETURN:
-                        SendMessageW(hWnd, WM_COMMAND, IDM_EVENT_DETAILS, 0);
+                #endif // LVN_ITEMACTIVATE
+                    {
+                        /* Get the index of the single focused selected item */
+                #ifdef LVN_ITEMACTIVATE
+                        LPNMITEMACTIVATE lpnmitem = (LPNMITEMACTIVATE)lParam;
+                        INT iItem = lpnmitem->iItem;
+                #else
+                        INT iItem = ListView_GetNextItem(hwndListView, -1, LVNI_FOCUSED | LVNI_SELECTED);
+                #endif // LVN_ITEMACTIVATE
+
+                        if (iItem != -1)
+                            SendMessageW(hWnd, WM_COMMAND, IDM_EVENT_DETAILS, (LPARAM)iItem);
                         break;
+                    }
```

Comment:
I would split this in 2 different distinct cases.
Having the LVN_ITEMACTIVATE and NM_DBLCLICK code intermingled is not making it any clearer, and you don't combine any big code blocks either.

Reply:
Actually I'm conditionally compiling the code here. If we support LVN_ITEMACTIVATE then I'm using it, otherwise it would fallback to handling NM_DBLCLK/NM_RETURN.

Reply:
Fixed.